spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,163 @@
+    end
+
+    it 'should build new user' do
+      expect(assigns(:user)).to be_a_new(User)
+    end
+  end
+
+  context 'POST#create' do
+    context 'when success' do
+      before(:each) { post :create, user: FactoryGirl.attributes_for(:user) }
+
+      it 'should redirect to ready polls' do
+        is_expected.to redirect_to(ready_polls_path)
+      end
+
+      it 'should create user', :skip_before do
+        expect { post :create, user: FactoryGirl.attributes_for(:user) }.to change { User.count }.by(1)
+      end
+
+      it 'should log in user' do
+        expect(session[:user_id]).to eq(assigns(:user).id)
+      end
+    end
+
+    context 'when not success' do
+      before(:each) { post :create, user: FactoryGirl.attributes_for(:user, username: '') }
+
+      it 'should render template new' do
+        is_expected.to render_template(:new)
+      end
+    end
+  end
+
+  context 'GET#edit' do
+    it 'should be success' do
+      session[:user_id] = user1.id
+      get :edit, id: user1.id
+
+      expect(response).to be_success
+    end
+
+    it 'should not build new user' do
+      get :edit, id: user1.id
+
+      expect(assigns(:user)).not_to be_a_new(User)
+    end
+
+    context 'when user not current user' do
+      it 'should redirect to login page if not logged in' do
+        session[:user_id] = nil
+        get :edit, id: user1.id
+
+        expect(response).to redirect_to login_path
+      end
+
+      it 'should reddirect to current user edit path' do
+        session[:user_id] = user2.id
+        get :edit, id: user1.id
+
+        expect(response).to redirect_to edit_user_path(user2)
+      end
+    end
+  end
+
+  context 'GET#show' do
+    before(:each) { get :show, id: user1.id }
+
+    it 'should be success' do
+      expect(response).to be_success
+    end
+
+    it 'should get correct user' do
+      expect(assigns(:user)).to eq(user1)
+    end
+  end
+
+  context 'PUT#update' do
+    context 'when input correct' do
+      it 'should redirect to show page' do
+        session[:user_id] = user1.id
+        expect(put :update, id: user1.id, user: { username: user1.username, email: user1.email }).to redirect_to(user_path)
+      end
+
+      it 'should change only user username' do
+        session[:user_id] = user2.id
+        put :update, id: user2.id, user: { username: 'newname', email: user2.email }
+        expect(assigns(:user).username).to eq("newname")
+      end
+
+      it 'should change user username' do
+        session[:user_id] = user2.id
+        put :update, id: user2.id, user: { username: user2.username, email: 'new@mail' }
+        expect(assigns(:user).email).to eq("new@mail")
+      end
+
+      it 'should not change anything if nothing changed' do
+        session[:user_id] = user1.id
+        put :update, id: user1.id, user: { username: user1.username, email: user1.email }
+        expect(assigns(:user)).to eq(user1)
+      end
+    end
+
+    context 'when input incorrect' do
```

Comment:
I used it `is_expected.to` and `before(:each)` in other tests, there I login as different users (thats why not `before(:each)`) and I just don't like how it looks: 
```
it 'should return our user' do
    subject
    expect(assigns(:user)).to eq(user1)
  end
```
You should scroll up to see what the subject is